packages/components/addon/components/hds/form/character-count/index.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+const ID_PREFIX = 'character-count-';
+const NOOP = () => {};
+
+export default class HdsFormCharacterCountIndexComponent extends Component {
+  // The current number of characters in the associated input
+  @tracked currentLength;
+  inputControl = document.getElementById(this.args.controlId);
```

Comment:
do you see any case in which the input element is added to the DOM _after_ the counter is injected? 🤔

Reply:
if used in a field component no. if used as a base element, hypothetically, yes, this can happen – not sure what we can do about that as we won't have an `id` to give the character count as a reference